### PR TITLE
Rich text posts

### PR DESCRIPTION
**Pull request by richardgirges**
This is the first draft of rich text editing with CK Editor.

Things to note:
* This PR contains PR #889 and should be rebased after 889 is merged

* There are no typings for the React CKEditor wrapper. The type check is failing right now because of this. [Lack of typings for CKEditor is a long-standing issue with no solid resolution as of yet](https://github.com/ckeditor/ckeditor5/issues/504). Please advise on how to approach

### CODE REVIEW

**Comment by CarsonF** [comment]
> * There are no typings for the React CKEditor wrapper. The type check is failing right now because of this. [Lack of typings for CKEditor is a long-standing issue with no solid resolution as of yet](https://github.com/ckeditor/ckeditor5/issues/504). Please advise on how to approach


We've got a few libraries we provide our own types for in the `typings` folder. It looks like there are types in DefinitelyTyped for the core of the library, so really we just need to provide connection points to this library. Should be trivial since both referenced are thin wrappers.

File: src/components/posts/PostListItemCard/PostListItemCard.tsx [hunk 1]
```diff
@@ -118,7 +119,9 @@ export const PostListItemCard: FC<PostListItemCardProps> = (props) => {
                 </Typography>
               </div>
             </div>
-            <Typography variant="body2">{props.body.value}</Typography>
+            <Typography variant="body2">
+              {parse(props.body.value || '')}
```
Comment (CarsonF) [inline src/components/posts/PostListItemCard/PostListItemCard.tsx:123]: We should probably memoize this, eh?

**Review by CarsonF — APPROVED** [review]
Looks good @richardgirges!

File: src/components/posts/PostListItemCard/PostListItemCard.tsx [hunk 2]
```diff
@@ -124,7 +127,7 @@ export const PostListItemCard: FC<PostListItemCardProps> = ({
                 </Typography>
               </div>
             </div>
-            <Typography variant="body2">{post.body.value}</Typography>
+            <Typography variant="body2">{postBody}</Typography>
```
Comment (CarsonF) [inline src/components/posts/PostListItemCard/PostListItemCard.tsx:130]: Currently this puts any html tags under a `<p>` tag which is invalid HTML.

**Review by CarsonF — CHANGES_REQUESTED** [review]
The CKE features should be limited to only those we can render. I'd remove quotes, tables, pics, and video.

We should style the output with `react-html-parser` and the display inside of the editor to be consistent with our theme.

**Comment by ParkeBrown** [comment]
@daniel-leod can you confirm that we still need to address the unresolved comment above?

**Comment by daniel-leod** [comment]
> @daniel-leod can you confirm that we still need to address the unresolved comment above?

@ParkeBrown I think the last commit would resolve that. This PR is ready for review.

**Comment by ParkeBrown** [comment]
Daniel believes he's covered the requested changes

**Comment by CarsonF** [comment]
I'd like to go with an editor that produces JSON instead, instead of html string. So rendering can be easily customized based on context.

**Pull request closed without merging**